editor/src/core/model/element-metadata-utils.ts

```diff
@@ -2100,6 +2104,38 @@ export const MetadataUtils = {
       })
     }
   },
+  getNonExpressionDescendants(
+    metadata: ElementInstanceMetadataMap,
+    pathTree: ElementPathTrees,
+    target: ElementPath,
+  ): Array<ElementInstanceMetadata> {
+    const childrenInMetadata = MetadataUtils.getChildrenOrdered(metadata, pathTree, target)
+    if (!isFeatureEnabled('Code in navigator')) {
+      return childrenInMetadata
+    }
+
+    return childrenInMetadata.flatMap((c) =>
```

Comment:
I think large hierarchies don't really exist, because expressions can not really have expression children (because then they are just parsed as a single expression), and we stop the recursion when we reach a non-expression.